src/main/java/org/symphonyoss/symphony/messageml/elements/DateSelector.java

```diff
@@ -13,29 +20,110 @@
  */
 public class DateSelector extends FormElement {
   public static final String MESSAGEML_TAG = "date-selector";
-
+  
+  private static final String PLACEHOLDER_ATTR = "placeholder";
+  
+  private static final String PRESENTATIONML_NAME_ATTR = "data-name";
+  private static final String PRESENTATIONML_PLACEHOLDER_ATTR = "data-placeholder";
+  
   private static final String PRESENTATIONML_TAG = "div";
-  private static final String MARKDOWN = "Date Selector";
+  private static final String MARKDOWN = "Date Selector:";
   private static final String CLASS_ATTR = "class";
 
   public DateSelector(Element parent) {
     super(parent, MESSAGEML_TAG);
   }
 
+  @Override
+  public void buildAll(MessageMLParser context, org.w3c.dom.Element element) throws InvalidInputException, ProcessingException {
+    switch (getFormat()) {
+      case MESSAGEML:
+        super.buildAll(context, element);
+        break;
+      case PRESENTATIONML:
+        buildElementFromPresentationML(context, element);
```

Comment:
It's buildFromDiv because it's the element used to build the object from, it's not the format.